src/pure-assert.js

```diff
@@ -55,40 +55,118 @@ export class UserError extends Error {
  * Throws a provided error if the second argument isn't trueish.
  * @param {Object} errorCls
  * @param {T} shouldBeTruthy
- * @param {string} message
+ * @param {string} opt_message
+ * @param {...*} var_args Arguments substituted into %s in the message
  * @return {T}
- * @throws {Error} when attribute values are missing or invalid.
+ * @throws {Error} when shouldBeTruthy is not truthy.
  */
-function pureAssertion(errorCls, shouldBeTruthy, message) {
+function pureAssertion(errorCls, shouldBeTruthy, opt_message, var_args) {
   // TODO: Support format strings.
-  if (!shouldBeTruthy) {
-    throw new errorCls(message);
+  if (shouldBeTruthy) {
+    return shouldBeTruthy;
   }
-  return shouldBeTruthy;
+
+  // Substitute provided values into format string in message
+  const message = Array.prototype.slice
+    .call(arguments, 3)
+    .reduce(
+      (msg, subValue) => msg.replace('%s', subValue),
+      opt_message || 'Assertion failed'
+    );
+
+  throw new errorCls(message);
```

Comment:
Just to clarify: having a special `UserError` class means relatively little because the errors are caught and rethrown all the time, sometimes by the code that we do not control, so the message testing is the only fully reliable way that we've found.

Reply:
Ack; expectation here was Bento code may throw `UserError` directly if we do not re-create `user.error()` structure. Will keep an eye on how this is used as error handling comes into place

Reply:
This is absolutely ok to throw a `UserError` by our code. Just making sure we're not overly reliant on it downstream. E.g. for a reporter `!(error instanceof UserError)` might still be a user error because someone has rethrown it weirdly.